tensorflow/core/lib/io/record_writer.cc

```diff
@@ -61,7 +61,7 @@ RecordWriter::RecordWriter(WritableFile* dest,
                            const RecordWriterOptions& options)
     : dest_(dest), options_(options) {
 #if defined(IS_SLIM_BUILD)
-  if (compression_type != compression::kNone) {
+  if (options.compression_type != RecordWriterOptions::NONE) {
```

Comment:
Thanks for catching this! But I believe this should be 

options.compression_type != compression::kNone